lib/src/new-select/NewSelect.jsx

```diff
@@ -0,0 +1,699 @@
+                }}
+                onClick={(event) => {
+                  event.stopPropagation();
+                }}
+                onMouseLeave={() => {
+                  changeVisualFocusedSuggIndex(-1);
+                }}
+              >
+                {searchable ? filteredOptions.map(mapOptionFunc) : options.map(mapOptionFunc)}
+              </OptionsList>
+            )}
+          </SelectContainer>
+          {!disabled && <Error backgroundType={backgroundType}>{error}</Error>}
+        </DxcSelect>
+      </ThemeProvider>
+    );
+  }
+);
+
+const sizes = {
+  small: "240px",
+  medium: "360px",
+  large: "480px",
+  fillParent: "100%",
+};
+
+const calculateWidth = (margin, size) =>
+  size === "fillParent"
+    ? `calc(${sizes[size]} - ${getMargin(margin, "left")} - ${getMargin(margin, "right")})`
+    : sizes[size];
+
+const DxcSelect = styled.div`
+  display: flex;
+  flex-direction: column;
+
+  width: ${(props) => calculateWidth(props.margin, props.size)};
+  margin: ${(props) => (props.margin && typeof props.margin !== "object" ? spaces[props.margin] : "0px")};
+  margin-top: ${(props) =>
+    props.margin && typeof props.margin === "object" && props.margin.top ? spaces[props.margin.top] : ""};
+  margin-right: ${(props) =>
+    props.margin && typeof props.margin === "object" && props.margin.right ? spaces[props.margin.right] : ""};
+  margin-bottom: ${(props) =>
+    props.margin && typeof props.margin === "object" && props.margin.bottom ? spaces[props.margin.bottom] : ""};
+  margin-left: ${(props) =>
+    props.margin && typeof props.margin === "object" && props.margin.left ? spaces[props.margin.left] : ""};
+`;
+
+const Label = styled.label`
+  color: ${(props) =>
+    props.disabled
+      ? props.backgroundType === "dark"
+        ? props.theme.disabledLabelFontColorOnDark
+        : props.theme.disabledLabelFontColor
+      : props.backgroundType === "dark"
+      ? props.theme.labelFontColorOnDark
+      : props.theme.labelFontColor};
+
+  font-family: ${(props) => props.theme.fontFamily};
+  font-size: ${(props) => props.theme.labelFontSize};
+  font-style: ${(props) => props.theme.labelFontStyle};
+  font-weight: ${(props) => props.theme.labelFontWeight};
+  line-height: 1.715em;
+`;
+
+const OptionalLabel = styled.span`
+  font-weight: ${(props) => props.theme.optionalLabelFontWeight};
+`;
+
+const HelperText = styled.span`
+  color: ${(props) =>
+    props.disabled
+      ? props.backgroundType === "dark"
+        ? props.theme.disabledHelperTextFontColorOnDark
+        : props.theme.disabledHelperTextFontColor
+      : props.backgroundType === "dark"
+      ? props.theme.helperTextFontColorOnDark
+      : props.theme.helperTextFontColor};
+
+  font-family: ${(props) => props.theme.fontFamily};
+  font-size: ${(props) => props.theme.helperTextFontSize};
+  font-style: ${(props) => props.theme.helperTextFontStyle};
+  font-weight: ${(props) => props.theme.helperTextFontWeight};
+  line-height: 1.5em;
+`;
```

Comment:
I think we should add a token for `helperTextLineHeight`, as in the rest of components.